### ISSUE TITLE
Maven host auth is not resolved with Renovate on Premises 1.6 

### ISSUE THREAD

**Issue by GreyTeardrop** [renovatebot/pro#170]
## Overview

As of version 1.6, Renovate on Prem isn't able to authenticate with the private Maven repository of my org. Version 1.5 and below worked fine. Opensource Renovate 27.31.10 with the same config seems to be able to authenticate as well.

## Details

We run Renovate on Prem (by the way, thank you for it, WhiteSource!) in Kubernetes, via a slightly modified Helm chart from this repo, and target our repositories on Github.com. Renovate's configuration looks like this
```js
module.exports = {
    "$schema": "https://docs.renovatebot.com/renovate-schema.json",
    "platform": "github",
    "autodiscover": false,
    "repositories": [
        "myorg/myrepos"
    ],
    "gitAuthor": "Renovate Bot <bot@renovateapp.com>",
    "hostRules": [
        {
            "matchHost": "myorg.jfrog.io",
            "username": "xxx",
            "password": "yyy"
        }
    ],
    "packageRules": [
        {
            "matchDatasources": [
                "maven"
            ],
            "registryUrls": [
                "https://myorg.jfrog.io/myorg/libs-release/",
                "https://repo.maven.apache.org/maven2"
            ]
        }
    ],
    "onboardingConfig": {
      "$schema": "https://docs.renovatebot.com/renovate-schema.json",
      "extends": [
        "github>myorg/renovate"
      ]
    }
}
```

Right after I've upgraded to 1.6.0, I've noticed that Renovate has closed all PRs updating our internal libs that are only available in myorg.jfrog.io. As far as I can see, Renovate still loads `hostRules` as expected:
```
DEBUG: Adding password authentication for myorg.jfrog.io to hostRules
```
however, it doesn't seem to apply them
```
DEBUG: Dependency lookup unauthorized. Please add authentication with a hostRule (repository=myorg/myrepo)
       "failedUrl": "https://myorg.jfrog.io/myorg/libs-release/ch/qos/logback/logback-classic/maven-metadata.xml"
```

I ran Renovate OSS with the same configuration (replacing Github app auth with a personal token and adding `dryRun: true`) and it looks like both version 25.48.0 and version 27.31.10 successfully apply `hostRules` and can fetch updates from our internal repository

**Comment by noemif** [comment]
We've observed exactly the same thing with Docker auth (the company has a paid account, so we're not hitting the pull limits). Our issues started just after we upgraded to v1.6, too.

**Comment by rarkins** [comment]
I was able to reproduce the problem locally and then fix it in 1.6.2. I hope that fully resolves it for you too, and please accept our apologies for the inconvenience caused.

If confirmed resolved, please close this issue.

**Comment by noemif** [comment]
Solved it for us too, thank you !
